### PR TITLE
Backport new integration tests to the 2.6.0 branch.

### PR DESCRIPTION
**Pull request by rjeberhard**
I made a small number of changes to get this to work with 2.6.0, but more importantly, I cut (for now) tests that used model-in-image domains.

The easiest way to review this might be to compare the `new-integration-tests-2.6.0` branch against `develop` and focus on the new-integration-tests directory.

Tests passed except for the ItPodsRestart test that failed with an assertion that should have passed: https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/338/

### CODE REVIEW

**Comment by rjeberhard** [comment]
Some more details:
- Operator 2.6.0 will not support model-in-image
- Operator 2.6.0 will not support work related to an introspect version.

Tests removed
- Because they relate to non-supported features:
* ItIntrospectVersion
* ItMiiAddCluster
* ItMiiChangeAdminCredentials
* ItMiiConfigMap
* ItMiiConfigMapOverride
* ItMiiDomain
* ItMiiSample
* ItScaleMiiDomainNginx

- Because the tests use a model-in-image domain -- we could rewrite alternate versions of these tests:
* ItMonitoringExporter
* ItSessionMigration

File: new-integration-tests/src/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+* assertions must have the message
+* must use our logging, timer, actions, assertions
+* actions must use api, not spawn kubectl process
+* diagnostics!!! 
+* document how we inject the k8s cluster
+* document how tags are to be used
+* tests must not assume anything - they need to set up for themselves
+* tests do not need to clean/delete (unless that is part of the actual test) 
+  since we will just throw away the cluster at the end anyway
+* no need for cleanup script - since every test will have a new cluster
```
Comment (vanajamukkara) [inline new-integration-tests/src/README.md:16]: this is true only for kind cluster :-)

Comment (rjeberhard) [reply new-integration-tests/src/README.md:16]: Should I change something here? ... I didn't write this. :)

Comment (vanajamukkara) [reply new-integration-tests/src/README.md:16]: oh ok, leave it then ... may be this was written thinking we will use ephemeral clusters on Jenkins

**Pull request merged**